scheduler/pkg/store/experiment/db_test.go

```diff
@@ -268,6 +321,112 @@ func TestSaveAndRestore(t *testing.T) {
 	}
 }
 
+func TestSaveAndRestoreDeletedExperiments(t *testing.T) {
+	g := NewGomegaWithT(t)
+	type test struct {
+		name        string
+		experiments []*Experiment
+		withTTL     bool
+	}
+
+	getStrPtr := func(val string) *string { return &val }
+
+	tests := []test{
+		{
+			name: "deleted experiment with ttl",
+			experiments: []*Experiment{
+				{
+					Name:         "test-with",
+					ResourceType: ModelResourceType,
+					Default:      getStrPtr("model1"),
+					Candidates: []*Candidate{
+						{
+							Name:   "model1",
+							Weight: 50,
+						},
+						{
+							Name:   "model2",
+							Weight: 50,
+						},
+					},
+					KubernetesMeta: &KubernetesMeta{
+						Namespace:  "default",
+						Generation: 2,
+					},
+					Deleted: true,
+				},
+			},
+			withTTL: true,
+		},
+		{
+			name: "deleted experiment without ttl",
+			experiments: []*Experiment{
+				{
+					Name:         "test-without",
+					ResourceType: ModelResourceType,
+					Default:      getStrPtr("model1"),
+					Candidates: []*Candidate{
+						{
+							Name:   "model1",
+							Weight: 50,
+						},
+						{
+							Name:   "model2",
+							Weight: 50,
+						},
+					},
+					KubernetesMeta: &KubernetesMeta{
+						Namespace:  "default",
+						Generation: 2,
+					},
+					Deleted: true,
+				},
+			},
+			withTTL: false,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			path := fmt.Sprintf("%s/db", t.TempDir())
+			logger := log.New()
+			edb, err := newExperimentDbManager(getExperimentDbFolder(path), logger)
+			g.Expect(err).To(BeNil())
+			for _, exp := range test.experiments {
```

Comment:
Not a strong opinion either way, but a thought: Can we skip any experiments that have `Deleted: false` here? The assertions wouldn't hold for them. I know the test name in itself should prevent someone from adding such a test case but....

Reply:
I've added an expectation that they're deleted instead.